### ISSUE TITLE
avatars: clearly show which avatar you are holding

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/limelight#206]
it should be always easy to see which avatar you are holding. UpStage has the "mirror", in which you see the icon of the av you're holding. it could be the same, or something like the one you are holding comes to the top of the list & has a black border or is highlighted in some way.

**Comment by vickis** [comment]
sorry 
i am pretty sure i used the password i had - but was told it was wrong ....
then when i used the forgot password link got the something went wrong email again...

**Comment by vickis** [comment]
ok i figured it out the isue is firefox on the mac

**Comment by vickis** [comment]
in hen test stage can see that i am holding the car av because it is greyed out ...
on chrome

**Comment by vickis** [comment]
in the walkthrough stage it shows as a crossed out circle if i try and reselect the avatar that i was initially holding 
however this means that because i tried to hold one after the other - i cannot now select and drop the leaf gif av ....

![screen shot 2017-09-11 at 11 49 05 am](https://user-images.githubusercontent.com/4249450/30254094-3f89b3a8-96e7-11e7-85af-f66ab8ed7827.png)

**Comment by vickis** [comment]
ooops :)

**Comment by helenvarley** [comment]
when i click on one avatar, that icon enlarges to take up all the space of the avatar palette. i guess this indicates that that is the avatar i'm holding, however it's not great to then not be able to see any of the rest of the avatars available. it won't be a good solution for quickly swapping between avatars, as we often need to do. i would prefer a larger palette & something more like the design mockup that i made some time ago  (attached f.y.i.)
[tool palettes.pdf](https://github.com/upstage-org/limelight/files/1293749/tool.palettes.pdf)

**Comment by helenvarley** [comment]
also the avatar icon is distorted to fill the available space - i.e. squished short & wide, which doesn't look great.

**Comment by rohrlach** [comment]
Can you provide Screenshots?

**Comment by helenvarley** [comment]
![screenshot at 2017-09-11 22 53 14](https://user-images.githubusercontent.com/3437324/30296319-0b859434-9744-11e7-9b2d-4ae605e6a790.png)

**Comment by rohrlach** [comment]
@JoeIceCream please liaise with @helenvarley and @vickis to effectively resolve this issue

**Comment by helenvarley** [comment]
i have said before that i think we need the tool palettes to be bigger; it should be as if they are layered over the chat - not more than the top third, i would say - so that there is adequate space to see at least some of the available media, & have some other tool buttons. (see my mock-up). the "mirror", or display of the avatar that you're holding, should just be one part of this. often we want to switch quickly between a couple of avatars, & having to do a lot of scrolling will significantly slow that down.

**Comment by vickis** [comment]
yes agree - do ned to have as much of the palette on hand as possible - and if it is click in place and click away that will mean can access it and still also not lose what is happening in chat
thanks team :)

**Comment by rohrlach** [comment]
@JoeIceCream can you modify the entire avatar tab such that the mirror is displayed above (Or blank if nothing is held) and the avatar selection buttons are below it.

It's fine if you have to extend the height of the tabs to achieve this.

**Comment by rohrlach** [comment]
@helenvarley I'm going to raise a new issue for a toolbox rework I have in mind.

**Comment by vickis** [comment]
sounds interesting! :)

**Comment by JoeIceCream** [comment]
@helenvarley @vickis Hi Helen and Vicki, I've completed fixing the distortion in the mirror but I'd also like to enable performers to change the size of the avatar that is placed in the stage. Where would you like this action (if at all)? I can do it within the stage or where you assign the avatar initially. what would you prefer?

The other option is that I set a default size for an avatar, and all avatar's will have the same height (but with varying width).

**Comment by helenvarley** [comment]
@JoeIceCream - it would be AWESOM to be able to change the size of the avatar on the stage. if you look at the mock-up of the tool palettes that i made, you can see that i imagined a slider in the tool palette that would allow you to increase or decrease the size of the avatar you are holding (i also imagined being able to adjust opacity, flip & rotate - but i'll stop now or i'll get in trouble from paul for putting too many things into one issue ;) )
[tool palettes.pdf](https://github.com/upstage-org/limelight/files/1340577/tool.palettes.pdf)

**Comment by rohrlach** [comment]
Resizing the avatar on the stage isn't part of this issue. @JoeIceCream or @helenvarley you should open a new issue for this.

**Comment by helenvarley** [comment]
what did i say! :D

**Comment by vickis** [comment]
oh awesome 
we do need those new issues in there (sounds exciting and super great if achievable!
wahoo!

**Comment by JoeIceCream** [comment]
Are we okay to close this now @rohrlach, @vickis, @helenvarley ?

**Comment by helenvarley** [comment]
yes, but ... it's unnecessary for this much space to be give to the "mirror" - there's a lot of wasted space either side. can we not make the mirror just the size of the avatar icon, & use the rest of the space for other controls?
 i'm also concerned about the limited number of avatars that will be visible without scrolling; i guess this is a new issue, but it should be comparable with how the backdrops are working now.
i guess these are new issues ...

**Comment by helenvarley** [comment]
this issue can't be closed yet as it is not easy to see which avatar you're holding, since the "mirror" is below the scroll of the palette. the palette needs to take more space, in order to show everything in it without scrolling. i guess that's a separate issue ...
